### PR TITLE
Merge fontdata branch into main

### PR DESCRIPTION
**Pull request by jahndan**
not much to say, this is the base we'll be building on for now

this branch currently includes the basic fontfile splitting script

### CODE REVIEW

**Comment by jahndan** [comment]
@AitanSingh Comment on anything that stands out that could be improved
- basic repo structure
- filenames I chose for characters
  - I think the filenames make a lot of sense for the jp characters I did label (pokedollar might be literal yen)
  - I don't feel great about "dot" (as in dot product; there is a "cross"), but "interpunct" feels too obscure
  - I abbreviated the names for quotation marks for the sake of brevity but it might be unclear
  - "target" might actually make more sense to be called "coin" b/c it definitely looks like a coin symbol
  - "music" instead of "note"?
  - "poke" and "mon" form a compound character (takes up two character spaces) that looks like PKMN
  - also if you want to label any of the ones I skipped, that's worth noting as well
- any other questionable choices you spot

(I just use Black formatter on default settings, so formatting isn't much of a concern)

also I'm not super concerned about the test cases in `readfont_tests` because the fontfile splitting script (as a whole) works correctly on the valid inputs that are fontfiles in the repo (any edge cases I didn't cover will only happen if someone is trying to repurpose code in there for other stuff, which shouldn't be necessary)
-- that said, if you're keen on it, go ahead (you'll definitely be capable with numpy arrays by the end of it)

File: src/font/data_00000002.font.png
Comment (jahndan) [inline src/font/data_00000002.font.png:1]: unlike the normal font, this uses 3 colors for the characters
the normal font uses black as the transparent background, but this uses white as the transparent background—black is actually in the image data for this

**Review by jahndan — COMMENTED** [review]
only issue I spotted is the palette mismatch between normal and bold

File: src/font/readfont_index.py [hunk 1]
```diff
@@ -0,0 +1,709 @@
...
+            "en.angry",
+            "",
+            "",
+            "",
+            "",
+            "",
+            "en.poke",  # PK
+        ],
+        [
+            "en.mon",  # MN
```
Comment (AitanSingh) [inline src/font/readfont_index.py]: is MN being used as mon shortened? If that's the case I think its pretty unclear.

File: src/font/readfont_index.py [hunk 2]
```diff
@@ -0,0 +1,709 @@
...
+            "en.target",  # might be coins
+            "en.circle",
+            "en.square",
+            "en.triangle",
+            "en.rhombus",  # diagonal square
+            # other symbols
+            "en.address",  # at symbol @
+        ],
+        [
+            "en.note",  # musical note
```
Comment (AitanSingh) [inline src/font/readfont_index.py]: music would probably make a bit more sense here

File: src/font/readfont_index.py [hunk 3]
```diff
@@ -0,0 +1,709 @@
...
+            "en.colon",
+            "en.semicolon",
+            # playing card suits
+            "en.spade",
+            "en.club",
+            "en.heart",
+            "en.diamond",
+            # miscellaneous shapes
+            "en.star",  # 5 point star
+            "en.target",  # might be coins
```
Comment (AitanSingh) [inline src/font/readfont_index.py]: this makes more sense as coin

File: src/font/readfont_index.py [hunk 4]
```diff
@@ -0,0 +1,709 @@
...
+            "",
+            "en.pokemoney",  # pokédollar symbol
+            "",
+            "",
+            "en.exclaim",  # exclamation mark
+            "en.question",  # question mark
+            "en.comma",
+            "en.period",  # full stop
+            "en.ellipsis",  # ...
+            "en.dot",  # interpunct •
```
Comment (AitanSingh) [inline src/font/readfont_index.py:543]: Maybe to not get confused with dot product we could call it mole?

**Review by AitanSingh — COMMENTED** [review]
Only comments I had were in readfont_index.py

File: src/font/readfont_index.py (hunk 1 above)
Comment (jahndan) [reply src/font/readfont_index.py]: these two characters display in-game as PK and MN because this is the abbreviation gamefreak uses for "PKMN Trainer" (pokemon trainer) or other bits of text where "Pokemon" would take up too many characters

in-game they display like this (on one line as small capitals)
```
P M
 K N
```
